test/ShapeCrawler.Tests.Unit/ShapeCollectionTests.cs

```diff
@@ -415,6 +415,32 @@ public void AddPicture_too_large_adds_svg_picture()
         pres.Validate();
     }
 
+    [Test]
+    [Explicit]
+    public void AddPicture_svg_with_text_matches_reference()
+    {
+        // Arrange
+
+        // This presentation contains the same SVG we're adding below, manually
+        // dragged in while running PowerPoint
+        var pres = new Presentation(StreamOf("055_svg_with_text.pptx"));
+        var shapes = pres.Slides[0].Shapes;
+        var expected = shapes.GetByName<IPicture>("Original");
+
+        var image = TestHelper.GetStream("1x1.svg");
+        image.Position = 0;
+
+        // Act
+        shapes.AddPicture(image);
+        pres.Validate();
+        var tempdir = Environment.GetEnvironmentVariable("TEMP") ?? throw new ApplicationException("TEMP directory not found");
+        pres.SaveAs($"{tempdir}\\AddPicture_svg_with_text_matches_reference.pptx");
+
+        // Assert
+        // This test must be checked manually. Load up the pptx saved above, and visually inspect
+        // the images to ensure they are identical. 
```

Comment:
I familiar with this situation. Ideally, of course, it would be to convert the shape with SVG-content to an image and compare this _actual_ image to the _expected_ image in some way.

I need some time to come up with at least some first step. I will back...

Reply:
Indeed. So when I tried this with the tools at hand, the Svg library, it happily renders the SVG-content perfectly. Thus there is a subtle difference in how PowerPoint client renders svg versus how Svg library does so. I'm hesitant to accuse PowerPoint of having a bug, but certainly I'd posit Svg's way is BETTER.

A much hackier way would be to examine the SVG-content of the inserted image and verify that there is no whitespace between `<text>` and `<tbody>`. At least that's something.

Reply:
The only way to get the result of the PowerPoint client render in a unit test is probably to use `Interop.PowerPoint` (COM). I've tried this in unit tests in production, and I have to say, it's horrible 😒. `Interop.PowerPoint` is really not suited for automation in backend code.

For now, that leaves your suggestion: do a comparison based on internal content. At my main job, we do similar things when checking the result of docx to HTML conversion. These tests are certainly far from ideal; Resilience To Refactoring is very low since they depend on implementation details.

You may try such a test. I hope there will be few such tests in the future.

Reply:
Ok, so done. See what you think!